pygments/lexers/markup.py

```diff
@@ -763,3 +765,632 @@ def _handle_cssblock(self, match):
+    }
+
+    tokens = {
+        'root': [
+            # Redirects
+            (r"""(?x)
+                (\A\s*?)
+                (\#REDIRECT:?) # may contain a colon
+                (\s+)
+                (\[\[)
+                    ([{}]+?)
+                    (?: (\#)([^#]*?) )?
+                    (?: (\|)([^\n|]*) )?
+                (\]\])
+                (\s*?\n)
+            """.format(title_char),
+             bygroups(Whitespace, Keyword, Whitespace, Punctuation, Name.Tag, Punctuation, Name.Label,
+                      Punctuation, Text, Punctuation, Whitespace)),
+            # Subheadings
+            (r'^(={2,6})(.+?)(\1)(\s*$\n)',
+             bygroups(Generic.Subheading, Generic.Subheading, Generic.Subheading, Whitespace)),
+            # Headings
+            (r'^(=.+?=)(\s*$\n)',
+             bygroups(Generic.Heading, Whitespace)),
+            # Double-slashed magic words
+            (words(double_slashes_i), Name.Function.Magic),
+            (words(double_slashes, prefix='(?-i:', suffix=')'), Name.Function.Magic),
+            # Raw URLs
+            (r'\b(?:{}){}{}*'.format('|'.join(protocols),
+             link_address, link_char_class), Name.Label),
+            # Magic links
+            (r"""(?x-i:
+                \b(?:RFC|PMID) {nbsp_char}+ [0-9]+\b
+                |
+                \bISBN {nbsp_char}
+                (?: 97[89] {nbsp_dash}? )?
+                (?: [0-9] {nbsp_dash}? ){{9}}
+                [0-9Xx]\b
+            )""".format(nbsp_char=nbsp_char, nbsp_dash=f'(?:-|{nbsp_char})'), Name.Function.Magic),
+            include('list'),
+            include('inline'),
+            include('text'),
+        ],
+        'list': [
+            # Description lists
+            (r'^;', Keyword, 'dt'),
+            # Ordered lists, unordered lists and indents
+            (r'^[#:*]+', Keyword),
+            # Horizontal rules
+            (r'^-{4,}', Keyword),
+        ],
+        'inline': [
+            # Signatures
+            (r'~{3,5}', Keyword),
+            # Entities
+            include('entity'),
+            # Bold & italic
+            (r"('')(''')(?!')", bygroups(Generic.Emph,
+             Generic.Strong), 'inline-italic-bold'),
+            (r"'''(?!')", Generic.Strong, 'inline-bold'),
+            (r"''(?!')", Generic.Emph, 'inline-italic'),
+            # Comments & parameters & templates
+            include('replaceable'),
+            # Media links
+            (
+                r"""(?x)
+                (\[\[)
+                    (File|Image) (:)
+                    ([{}]*)
+                    (?: (\#) ([{}]*?) )?
+                """.format(title_char, f'{title_char}#%'),
+                bygroups(Punctuation, Name.Namespace,  Punctuation,
+                         Name.Tag, Punctuation, Name.Label),
+                'medialink-inner'
+            ),
+            # Wikilinks
+            (
+                r"""(?x)
+                (\[\[)(?!{}) # Should not contain URLs
```

Comment:
Ah, here `{}` is replaced with a list of url protocols by the preceding `format()`, like `(\[\[)(?!bitcoin:|ftp://|ftps://)`.

This is because an internal link with a valid url inside is not considered an internal link. It should be an external one.